From aa7e997d0c46a486038a5709cf6aaae93cacd6c1 Mon Sep 17 00:00:00 2001
From: Paul Naoumenko <paulx.naoumenko@intel.com>
Date: Fri, 18 Oct 2013 20:19:17 +0200
Subject: [PORT FROM MAIN] wifi: Add a property to select SoftAP channel

BZ: 94270

Add a property to be able to choose the channel
to be used by SoftAP.

Feature   : CWS.WIFI-Common

Change-Id: If8636f99780861950dfe8fd3c110279a26f628dd
Orig-Change-Id: Ib57e69cf75c52e7a92815cb1cacf8ad674baa141
Category: device enablement
Domain: CWS.WIFI-Common
Origin: upstream
Upstream-Candidate: yes
Signed-off-by: Jean Trivelly <jean.trivelly@intel.com>
---
 SoftapController.cpp |    9 ++++++++-
 1 files changed, 8 insertions(+), 1 deletions(-)

diff --git a/SoftapController.cpp b/SoftapController.cpp
index 995e1de..5fc4f72 100644
--- a/SoftapController.cpp
+++ b/SoftapController.cpp
@@ -35,6 +35,7 @@
 
 #define LOG_TAG "SoftapController"
 #include <cutils/log.h>
+#include <cutils/properties.h>
 #include <netutils/ifc.h>
 #include <private/android_filesystem_config.h>
 #include "wifi.h"
@@ -165,13 +166,19 @@ int SoftapController::setSoftap(int argc, char *argv[]) {
 
     if (!strcasecmp(argv[4], "hidden"))
         hidden = 1;
-
+     
     if (argc >= 5) {
         channel = atoi(argv[5]);
         if (channel <= 0)
             channel = AP_CHANNEL_DEFAULT;
+    } else {
+        char ap_channel_s[10];
+        property_get("wifi.ap.channel", ap_channel_s, "6");
+        channel = atoi(ap_channel_s);
     }
 
+
+
     asprintf(&wbuf, "interface=%s\ndriver=nl80211\nctrl_interface="
             "/data/misc/wifi/hostapd\nssid=%s\nchannel=%d\nieee80211n=1\n"
             "hw_mode=g\nignore_broadcast_ssid=%d\n",
-- 
1.7.4.1

